### PR TITLE
[Improvement][Security]Add CSRF protection using spring security(issue-12931)

### LINKED ISSUE #12931: [Improvement][Security] Add CSRF protection
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

* Currently, we do not have CSRF protecting for form submission and we haven't enabled `CSRF` of Spring Security. We could enable it and improve the front end at the same time by adding `CSRF-token` to protect `DS` from `CSRF` attack.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hdygxsj**
## Purpose of the pull request

close #12931

In contrast to pr #13164, which is a csrf defense implemented through spring security, all requests will go through the csrfFilter and load the token in the cookie to compare to the token in the header or paramter

## Brief change log

* spring security dependencies were introduced
* Added a class LoginCsrfTokenRepository that implements CsrfTokenRepository
* Changed the login api.Upon login, the csrf token is generated through LoginCsrfTokenRepository and returned to the front-end
* The front end saves the csrf token returned by the login request to pinia.
* Whenever the front-end makes an http request, load the csrf token from pinia and place it on the X-XSRF-TOKEN of the http request header

## Verify this pull request

All api test cases are modified to ensure that they pass csrf validation

### CODE REVIEW

File: dolphinscheduler-api-test/dolphinscheduler-api-test-case/src/test/java/org/apache/dolphinscheduler/api.test/utils/RequestClient.java [hunk 1]
```diff
@@ -24,16 +24,11 @@
...
 import java.util.Objects;
 
-import org.testcontainers.shaded.okhttp3.FormBody;
-import org.testcontainers.shaded.okhttp3.Headers;
-import org.testcontainers.shaded.okhttp3.MediaType;
-import org.testcontainers.shaded.okhttp3.OkHttpClient;
-import org.testcontainers.shaded.okhttp3.Request;
-import org.testcontainers.shaded.okhttp3.RequestBody;
-import org.testcontainers.shaded.okhttp3.Response;
+import org.testcontainers.shaded.okhttp3.*;
```
Comment (SbloodyS) [inline dolphinscheduler-api-test/dolphinscheduler-api-test-case/src/test/java/org/apache/dolphinscheduler/api.test/utils/RequestClient.java]: Please roll back these unnessnary changes.

**Review by SbloodyS — CHANGES_REQUESTED** [review]

File: dolphinscheduler-api-test/dolphinscheduler-api-test-case/src/test/java/org/apache/dolphinscheduler/api.test/utils/RequestClient.java (hunk 1 above)
Comment (hdygxsj) [reply dolphinscheduler-api-test/dolphinscheduler-api-test-case/src/test/java/org/apache/dolphinscheduler/api.test/utils/RequestClient.java]: done

File: tools/dependencies/known-dependencies.txt [hunk 2]
```diff
@@ -313,8 +314,11 @@ spring-jcl-5.3.22.jar
 spring-jdbc-5.3.22.jar
 spring-ldap-core-2.4.1.jar
 spring-retry-1.3.3.jar
+spring-security-config-5.7.3.jar
```
Comment (kezhenxu94) [inline tools/dependencies/known-dependencies.txt:317]: Please also add these dependencies' license to this file 

https://github.com/apache/dolphinscheduler/blob/dev/dolphinscheduler-dist/release-docs/LICENSE

**Comment by hdygxsj** [comment]
Some changes have been made to the implementation of this pr
1.  Added an api for obtaining the csrf token.  When the api is requested, the csrf token will also be added to the cookie
2.  The front-end will request this api to obtain the token and store it in pinia before login
3.  The front-end will fetch the token from the pinia and add it to the Http header when making an Http request
4.  The http request will be intercepted by spring security's Csrf filter and compare the token in the http header to the token in the cookie, unless the http request header contains a token.  http requests that contain tokens are validated by the token interceptor

**Comment by hdygxsj** [comment]
Can you take a code review when you free @SbloodyS @kezhenxu94

**Comment by hdygxsj** [comment]
Is there anything else I need to fix

**Comment by hdygxsj** [comment]
@EricGao888 can you take a code review when you free

**Comment by EricGao888** [comment]
> @EricGao888 can you take a code review when you free

Sure, will take a look later today.

**Pull request closed without merging**